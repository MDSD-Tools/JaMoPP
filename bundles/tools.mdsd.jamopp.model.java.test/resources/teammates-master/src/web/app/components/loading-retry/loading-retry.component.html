<div *ngIf="shouldShowRetry; else content" class="text-center">
  <p class="text-muted m-1">{{message}}</p>
  <button (click)="retryEvent.emit()" class="btn btn-primary"><i class="fas fa-redo" aria-hidden="true"></i>&nbsp;Retry</button>
</div>

<ng-template #content>
  <ng-content></ng-content>
</ng-template>
