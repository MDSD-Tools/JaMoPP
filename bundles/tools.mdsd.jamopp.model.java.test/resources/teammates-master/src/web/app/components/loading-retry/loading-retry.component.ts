import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';

/**
 * Retry button for when loading fails.
 */
@Component({
  selector: 'tm-loading-retry',
  templateUrl: './loading-retry.component.html',
  styleUrls: ['./loading-retry.component.scss'],
})
export class LoadingRetryComponent implements OnInit {

  @Input() message: string = '';

  @Input() shouldShowRetry: boolean = false;

  @Output() retryEvent: EventEmitter<any> = new EventEmitter<any>();

  constructor() { }

  ngOnInit(): void {
  }

}
