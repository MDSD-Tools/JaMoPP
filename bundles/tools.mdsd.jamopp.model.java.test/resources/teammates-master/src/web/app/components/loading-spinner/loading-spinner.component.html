<div class="loading-container">
  <div class="row spinner-border text-primary" role="status"></div>
  <div class="row text-secondary">Loading...</div>
</div>
