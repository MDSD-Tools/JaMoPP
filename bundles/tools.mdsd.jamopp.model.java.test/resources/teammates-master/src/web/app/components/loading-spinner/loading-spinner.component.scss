.loading-container {
  margin: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.spinner-border {
  width: 3rem;
  height: 3rem;
  display: inline-table;
}
