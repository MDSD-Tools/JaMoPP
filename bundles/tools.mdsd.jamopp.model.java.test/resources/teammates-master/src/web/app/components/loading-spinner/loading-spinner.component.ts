import { Component } from '@angular/core';

/**
 * Loading spinner to show when waiting for request.
 */
@Component({
  selector: 'tm-loading-spinner',
  templateUrl: './loading-spinner.component.html',
  styleUrls: ['./loading-spinner.component.scss'],
})
export class LoadingSpinnerComponent {

  constructor() { }

}
