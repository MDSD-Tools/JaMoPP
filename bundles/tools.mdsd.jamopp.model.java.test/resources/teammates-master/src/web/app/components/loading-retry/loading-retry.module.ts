import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { LoadingRetryComponent } from './loading-retry.component';

/**
 * Module for the retry button
 */
@NgModule({
  declarations: [LoadingRetryComponent],
  imports: [
    CommonModule,
  ],
  exports: [LoadingRetryComponent],
})
export class LoadingRetryModule { }
