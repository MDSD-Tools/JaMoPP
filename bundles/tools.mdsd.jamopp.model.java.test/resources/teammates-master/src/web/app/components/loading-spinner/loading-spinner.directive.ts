import {
  ComponentFactory,
  ComponentFactoryResolver,
  ComponentRef,
  Directive,
  Input,
  TemplateRef,
  ViewContainerRef,
} from '@angular/core';
import { LoadingSpinnerComponent } from './loading-spinner.component';

/**
 * Directive for loading spinner component
 */
@Directive({
  selector: '[tmIsLoading]',
})
export class LoadingSpinnerDirective {
  loadingComponent: ComponentRef<LoadingSpinnerComponent>;
  loadingFactory: ComponentFactory<LoadingSpinnerComponent>;

  constructor(private templateRef: TemplateRef<any>,
              private viewContainer: ViewContainerRef,
              private componentFactoryResolver: ComponentFactoryResolver) {
    this.loadingFactory = this.componentFactoryResolver.resolveComponentFactory(LoadingSpinnerComponent);
    this.loadingComponent = this.viewContainer.createComponent(this.loadingFactory);
  }

  @Input() set tmIsLoading(loading: boolean) {
    this.viewContainer.clear();

    if (loading) {
      this.loadingComponent = this.viewContainer.createComponent(this.loadingFactory);
    } else {
      this.viewContainer.createEmbeddedView(this.templateRef);
    }
  }
}
