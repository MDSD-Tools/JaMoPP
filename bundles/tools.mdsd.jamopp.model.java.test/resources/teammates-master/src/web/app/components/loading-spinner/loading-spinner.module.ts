import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { LoadingSpinnerComponent } from './loading-spinner.component';
import { LoadingSpinnerDirective } from './loading-spinner.directive';

/**
 * Module for the loading spinner
 */
@NgModule({
  declarations: [LoadingSpinnerDirective, LoadingSpinnerComponent],
  imports: [
    CommonModule,
  ],
  exports: [LoadingSpinnerDirective],
  entryComponents: [LoadingSpinnerComponent],
})
export class LoadingSpinnerModule { }
