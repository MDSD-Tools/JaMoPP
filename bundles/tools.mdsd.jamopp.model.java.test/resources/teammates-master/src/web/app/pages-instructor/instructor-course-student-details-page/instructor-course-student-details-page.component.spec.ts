import { HttpClientTestingModule } from '@angular/common/http/testing';
import { Component, Input } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { Gender, JoinState, Student, StudentProfile } from '../../../types/api-output';
import { LoadingRetryModule } from '../../components/loading-retry/loading-retry.module';
import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
import {
  InstructorCourseStudentDetailsPageComponent,
} from './instructor-course-student-details-page.component';

@Component({ selector: 'tm-student-profile', template: '' })
class StudentProfileStubComponent {
  @Input() studentProfile: StudentProfile | undefined;
  @Input() studentName: string = '';
  @Input() photoUrl: string = '/assets/images/profile_picture_default.png';
  @Input() hideMoreInfo: boolean = false;
}
@Component({ selector: 'tm-course-related-info', template: '' })
class CourseRelatedInfoStubComponent {
  @Input() student: Student = {
    email: '',
    courseId: '',
    name: '',
    comments: '',
    teamName: '',
    sectionName: '',
    joinState: JoinState.JOINED,
  };
}
@Component({ selector: 'tm-more-info', template: '' })
class MoreInfoStubComponent {
  @Input() studentName: string = '';
  @Input() moreInfoText: string = '';
}

describe('InstructorCourseStudentDetailsPageComponent', () => {
  let component: InstructorCourseStudentDetailsPageComponent;
  let fixture: ComponentFixture<InstructorCourseStudentDetailsPageComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        InstructorCourseStudentDetailsPageComponent,
        StudentProfileStubComponent,
        CourseRelatedInfoStubComponent,
        MoreInfoStubComponent,
      ],
      imports: [
        HttpClientTestingModule,
        RouterTestingModule,
        LoadingSpinnerModule,
        LoadingRetryModule,
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(InstructorCourseStudentDetailsPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default fields', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with populated student', () => {
    component.student = {
      email: 'studentEmail@email.com',
      courseId: 'CS3281',
      name: 'firstName',
      comments: 'This is a comment',
      teamName: 'myTeam',
      sectionName: 'mySection',
      joinState: JoinState.JOINED,
    };
    component.studentProfile = {
      name: 'name',
      shortName: 'shortName',
      email: 'profileEmail@email.com',
      institute: 'NUS',
      nationality: 'Indian',
      gender: Gender.MALE,
      moreInfo: 'I have more info here',
    };
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
