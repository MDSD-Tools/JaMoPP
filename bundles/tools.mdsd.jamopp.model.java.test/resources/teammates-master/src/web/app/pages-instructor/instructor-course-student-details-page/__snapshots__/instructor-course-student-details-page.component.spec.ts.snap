// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructorCourseStudentDetailsPageComponent should snap with default fields 1`] = `
<tm-instructor-course-student-details-page
  courseId="undefined"
  hasStudentLoadingFailed="false"
  hasStudentProfileLoadingFailed="false"
  isStudentLoading={[Function Boolean]}
  isStudentProfileLoading={[Function Boolean]}
  photoUrl={[Function String]}
  route={[Function ActivatedRoute]}
  statusMessageService={[Function StatusMessageService]}
  studentEmail="undefined"
  studentProfileService={[Function StudentProfileService]}
  studentService={[Function StudentService]}
>
  <tm-loading-retry>
    
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
    
    
  </tm-loading-retry>
</tm-instructor-course-student-details-page>
`;

exports[`InstructorCourseStudentDetailsPageComponent should snap with populated student 1`] = `
<tm-instructor-course-student-details-page
  courseId="undefined"
  hasStudentLoadingFailed="false"
  hasStudentProfileLoadingFailed="false"
  isStudentLoading={[Function Boolean]}
  isStudentProfileLoading={[Function Boolean]}
  photoUrl={[Function String]}
  route={[Function ActivatedRoute]}
  statusMessageService={[Function StatusMessageService]}
  student={[Function Object]}
  studentEmail="undefined"
  studentProfile={[Function Object]}
  studentProfileService={[Function StudentProfileService]}
  studentService={[Function StudentService]}
>
  <tm-loading-retry>
    
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
    
    
  </tm-loading-retry>
</tm-instructor-course-student-details-page>
`;
